### PR TITLE
World_news show replace bootstrap with GDS

### PR DESCRIPTION
**Pull request by CL-GDS**
# What
* Dupe and Rename `show`
* Dupe and Rename `controller_test`
* Added `next_release` to `controller`
* Added `layout` to `controller`
* GDS to `show` 
# Why
* Show to GDS 
# Trello
https://trello.com/c/kNntszsl/51-world-location-news-show-page
# Pics
Screenshot 2023-03-17 at 14 03 49

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Comment by CL-GDS** [comment]
I did not create a helper for the tabs as it only appears on this page. Could be a extracted into helper refactor in the future

File: app/views/admin/world_location_news/show.html.erb [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 
-<%= content_tag_for(:div, @world_location_news) do %>
-  <div class="world-location-header">
-    <h1>
-      <span class="name"><%= @world_location_news.name %></span>
-    </h1>
-    <p>
-    <%= link_to "View on website", @world_location_news.public_url(cachebust_url_options) %>
-  </div>
+<%= link_to "View on website", @world_location_news.public_url(cachebust_url_options), class: "govuk-link" %>
```
Comment (davidgisbey) [inline app/views/admin/world_location_news/show.html.erb:5]: we should wrap this and everything below in 2/3 width please

File: app/views/admin/world_location_news/show.html.erb [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+      href: admin_world_location_news_path(@world_location_news),
+      current: true,
+    },
+    {
+      label: "Translations",
+      href: admin_world_location_news_translations_path(@world_location_news),
+    },
+    {
+      label: "Features (#{I18n.locale})",
+      href: features_admin_world_location_news_path(@world_location_news, locale: I18n.locale),
```
Comment (davidgisbey) [inline app/views/admin/world_location_news/show.html.erb:21]: So i think the behaviour of this is subtly different to the current implementation

```
  def world_location_news_tabs(world_location)
    tabs = {
      "Details" => admin_world_location_news_path(world_location),
      "Translations" => admin_world_location_news_translations_path(world_location),
      "Features (#{Locale.new(:en).native_language_name})" => features_admin_world_location_news_path(world_location, locale: I18n.default_locale),
    }
    world_location.non_english_translated_locales.each do |locale|
      tabs["Features (#{locale.native_language_name})"] = features_admin_world_location_news_path(world_location, locale: locale.code)
    end
    tabs
  end
```

This ☝️ creates a features tab for the English translation and then an additional features  tab for each non-english translation.

You could do something like

```
<%= render "components/secondary_navigation", {
  aria_label: "Document tabs",
  items: [
    {
      label: "Details",
      href: admin_world_location_news_path(@world_location_news),
      current: true,
    },
    {
      label: "Translations",
      href: admin_world_location_news_translations_path(@world_location_news),
    },
    {
      label: "Features (#{Locale.new(:en).native_language_name})",
      href: features_admin_world_location_news_path(@world_location_news, locale: I18n.locale),
    },
    world_location.non_english_translated_locales.map do |locale|
      {
        label: "Features (#{locale.native_language_name})",
        href: features_admin_world_location_news_path(world_location, locale: locale.code),
      },
    end
  ].compact
}%>

```

File: test/functional/admin/legacy_world_location_news_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,126 @@
+require "test_helper"
+
+class Admin::WorldLocationNewsControllerTest < ActionController::TestCase
+  setup do
+    login_as :writer
```
Comment (davidgisbey) [inline test/functional/admin/legacy_world_location_news_controller_test.rb:5]: we need to add the "Preview design system" permission to the user for the non-legacy tests

```
test/functional/admin/lworld_location_news_controller_test.rb:5

login_as_preview_design_system_user(:writer)
```

**Comment by davidgisbey** [comment]
Looking good. Just left a few comments. It'll need a lint too

**Comment by cbernardojr-gds** [comment]
@CL-GDS 
Hey!
Comments:
- Increase the space under the "View on website" link
- Add a caption over the title "World location"
- Move the Edit grey button above the text and below the tabs/subnav

Reference: https://gds-wh.herokuapp.com/v2/organisation/
Password: wh1t3h@11

**Comment by DilwoarH** [comment]
Is this ready to review or still draft?

**Comment by CL-GDS** [comment]
RFR

**Comment by DilwoarH** [comment]
Screenshot 2023-03-21 at 10 37 10 am

Just pulled this down and the spacing is still not great between `view on website` and the nav

**Comment by DilwoarH** [comment]
Screenshot 2023-03-21 at 10 46 25 am

@cbernardojr-gds - not sure if I agree with the positioning of the Edit button - it just looks out of place. Can we try something else?

File: app/views/admin/world_location_news/show.html.erb [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
+    <div class="govuk-grid-column-two-thirds">
+      <%= render "components/secondary_navigation", {
+        aria_label: "Document tabs",
+        items: secondary_navigation_tabs_items(current_user, @world_location_news, request.path)
+      } %>
+      <div class="govuk-!-margin-bottom-4">
+        <%= render "govuk_publishing_components/components/button", {
+          text: "Edit",
+          secondary_solid: true,
+          href: link_to("Edit", [:edit, :admin, @world_location_news])
```
Comment (DilwoarH) [inline app/views/admin/world_location_news/show.html.erb:18]: This is inserting a link within a link.

Screenshot 2023-03-21 at 10 52 58 am

**Comment by DilwoarH** [comment]
The spacing between the view on website link needs to be fix.

If you follow my suggestion [here](https://github.com/alphagov/whitehall/pull/7450#pullrequestreview-1350149642) - this will be fixed

File: app/views/admin/world_location_news/show.html.erb [hunk 5]
```diff
@@ -1,24 +1,25 @@
...
-    <%= tab_navigation_for(@world_location_news) %>
-
-    <div class="mission-statement lead">
+<div class="govuk-!-margin-bottom-8">
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <% content_for :page_title, @world_location_news.name %>
+      <% content_for :context, "World location" %>
+      <% content_for :title, @world_location_news.name %>
+      <% content_for :title_margin_bottom, 4 %>
```
Comment (DilwoarH) [inline app/views/admin/world_location_news/show.html.erb:7]: This should be on the top of the file

Comment (DilwoarH) [inline app/views/admin/world_location_news/show.html.erb:7]:
```
      <% content_for :page_title, @world_location_news.name %>
      <% content_for :context, "World location" %>
      <% content_for :title, @world_location_news.name %>
      <% content_for :title_margin_bottom, 4 %>
```

This should be on the top of the file

**Comment by CL-GDS** [comment]
RFR

**Review by davidgisbey — APPROVED** [review]

**Pull request merged**